### PR TITLE
🐛 Aml mlflow fix

### PR DESCRIPTION
**Pull request by trajepl**
## Describe your changes
For version azureml-evaluate-mlflow<=0.0.7, the default flaver name for huggingface transformers model is `hftransformer`.
But after that(>=0.08, currently the latest one is 0.0.25), the default name goes to `hftransformerv2`.

Also when it >=0.0.14, the version of mlflow[-skinny] will be forced to 2.3.1.

This PR used to reduce the version range for azureml-evaluate-mlflow and mlflow which is strong rely on mlflow[-skinny]==2.3.1.
If not, the pip resolver is always try to go through all the version and raise the following error:
image

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: test/requirements-test.txt [hunk 1]
```diff
@@ -16,11 +16,7 @@ pytorch_lightning
 tabulate
 torchvision
 datasets
-azureml-evaluate-mlflow
-mlflow
-# newest version mlflow-skinny==2.4.0 is not compatible with the test
-# issue: https://github.com/mlflow/mlflow/issues/8629
-mlflow-skinny==2.3.2
+azureml-evaluate-mlflow>=0.0.14
```
Comment (trajepl) [inline test/requirements-test.txt:19]: azureml-evaluate-mlflow will force the version of mlflow-skinny to 2.3.1. So I removed it.

**Review by leqiao-1 — APPROVED** [review]

File: olive/model/__init__.py [hunk 2]
```diff
@@ -551,7 +551,14 @@ def load_mlflow_model(self):
...
         with open(os.path.join(self.model_path, "MLmodel"), "r") as fp:
             mlflow_data = yaml.safe_load(fp)
-            hf_pretrained_class = mlflow_data["flavors"]["hftransformers"]["hf_pretrained_class"]
+            # default flavor is "hftransformersv2" name from azureml.evaluate.mlflow>=0.0.8
+            # "hftransformers" is the name from azureml.evaluate.mlflow<0.0.8
+            # TODO: let user specify flavor name if needed
+            flavors_names = ["hftransformersv2", "hftransformers"]
+            for flavors_name in flavors_names:
+                if flavors_name in mlflow_data["flavors"]:
+                    hf_pretrained_class = mlflow_data["flavors"][flavors_name]["hf_pretrained_class"]
```
Comment (guotuofeng) [inline olive/model/__init__.py:560]: if the first key doesn't exist, the KeyError will be raise.

Comment (trajepl) [reply olive/model/__init__.py:560]: flavors always exits.

Comment (guotuofeng) [reply olive/model/__init__.py:560]: hftransformersv2

Comment (trajepl) [reply olive/model/__init__.py]: Fixed.

File: olive/model/__init__.py [hunk 3]
```diff
@@ -551,7 +551,21 @@ def load_mlflow_model(self):
...
+            # "hftransformers" from azureml.evaluate.mlflow<0.0.8
+            # TODO: let user specify flavor name if needed
+            hf_pretrained_class = None
+            flavors_names = ["hftransformersv2", "hftransformers"]
+            for flavors_name in flavors_names:
+                if flavors_name in mlflow_data.get("flavors", {}):
+                    hf_pretrained_class = mlflow_data["flavors"].get(flavors_name, {}).get("hf_pretrained_class", None)
+                    break
+            if not hf_pretrained_class:
+                raise ValueError(
```
Comment (guotuofeng) [inline olive/model/__init__.py:564]:
```python
            for flavors_name in flavors_names:
                flavors_name_config = mlflow_data["flavors"].get(flavors_name)
                if flavors_name_config:
                    hf_pretrained_class = flavors_name_config["hf_pretrained_class"]
                    break

```

Comment (trajepl) [reply olive/model/__init__.py:564]: 'hf_pretrained_class' may not exist.

Comment (guotuofeng) [reply olive/model/__init__.py:564]: If not exist, it will raise KeyError

Comment (guotuofeng) [reply olive/model/__init__.py:564]: https://msdata.visualstudio.com/Vienna/_git/AzureMlCli?path=/src/azureml-evaluate-mlflow/azureml/evaluate/mlflow/hftransformers/__init__.py&version=GBmaster&line=788&lineEnd=789&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Comment (guotuofeng) [reply olive/model/__init__.py:564]: https://msdata.visualstudio.com/Vienna/_git/AzureMlCli?path=/src/azureml-evaluate-mlflow/azureml/evaluate/mlflow/hftransformers/__init__.py&version=GBmaster&line=856&lineEnd=858&lineStartColumn=1&lineEndColumn=75&lineStyle=plain&_a=contents

Comment (guotuofeng) [reply olive/model/__init__.py:564]: https://github.com/mlflow/mlflow/blob/master/mlflow/utils/model_utils.py

**Review by guotuofeng — APPROVED** [review]

**Pull request merged**